plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/SqlStandardAccessControl.java

```diff
@@ -317,7 +317,7 @@ public void checkCanAlterColumn(ConnectorSecurityContext context, SchemaTableNam
     @Override
     public void checkCanSetTableAuthorization(ConnectorSecurityContext context, SchemaTableName tableName, TrinoPrincipal principal)
     {
-        if (!isAdmin(context)) {
+        if (!isTableOwner(context, tableName)) {
```

Comment:
This is insecure. IMO it would introduce a vector attack for Trino that was removed in https://github.com/trinodb/trino/pull/10351 (it is on purpose not having all details).

It is not enough to see you can own the object, but we need to make sure that to what users we can grant it. See some thinking here https://github.com/trinodb/trino/pull/16691#pullrequestreview-1354756527

Reply:
Table owner can grant any permission to any user including the WITH GRANT OPTION, so I don't understand how changing ownership that actually add the privilege to DROP the table will add a new security risk.

Same for views. If the view security mode was set to DEFINER than all masking functions and filter rules will run on behalf of the new owner privileges, so if the new owner is blocked from viewing some specific sensitive columns,  all executed queries on the view will result with masked data.
Views with INVOKER security mode are executed with the privileges of the authenticated users running the queries - so no issue here

Reply:
>  if the new owner is blocked from viewing some specific sensitive columns

Yes. But access could be different and so it could happen that things that here not visible to original user are now accessible to that user by querying the view they created and altered (transferred ownerhsip).

Reply:
> But access could be different and so it could happen that things that here not visible to original user are now accessible to that user by querying the view they created and altered

You are right. The view owner can change the ownership to any privileged user, so we should allow it only for views created with INVOKER security mode

Reply:
> so we should allow it only for views created with INVOKER security mode

I am not a fan of such conditions. I would prefer to have secure security model instead. However I am not sure if it is doable in hive sql-standard.

Reply:
Views can be recreated by a user with proper privileges to become the new owner, so I can skip this change for views and keep it for tables only.